docs/ref/models/fields.txt

```diff
@@ -2266,6 +2201,113 @@ accepted by :class:`ForeignKey`, plus one extra argument:
 See :doc:`One-to-one relationships </topics/db/examples/one_to_one>` for usage
 examples of ``OneToOneField``.
 
+.. _lazy-relationships:
+
+Lazy relationships
+------------------
+
+Lazy relationships allow you to reference models using their names as strings
+or to create recursive relationships. You can use strings as the first
+argument in any relationship field to reference models lazily. The following
+types of lazy references are available:
+
+.. _recursive-relationships:
+
+Recursive
+~~~~~~~~~
+
+To define a relationship where a model references itself, use ``"self"`` as
+the first argument of the relationship field::
+
+    from django.db import models
+
+
+    class Manufacturer(models.Model):
+        suppliers = models.ManyToManyField("self", symmetrical=False)
+
+When used in an :ref:`abstract model <abstract-base-classes>`, the recursive
+relationship will point to concrete model subclasses.
+
+Relative
+~~~~~~~~
+
+If you need to create a relationship on a model that has not yet been defined,
+you can use the name of the model instead of the model object itself::
+
+    from django.db import models
+
+
+    class Car(models.Model):
+        manufacturer = models.ForeignKey(
+            "Manufacturer",
+            on_delete=models.CASCADE,
+        )
+        # ...
+
+
+    class Manufacturer(models.Model):
+        # ...
+        pass
+
+Relationships defined this way on :ref:`abstract models
+<abstract-base-classes>` are resolved when the model is subclassed as a
+concrete model and are not relative to the abstract model's ``app_label``:
+
+.. code-block:: python
+    :caption: ``products/models.py``
+
+    from django.db import models
+
+
+    class AbstractCar(models.Model):
+        manufacturer = models.ForeignKey("Manufacturer", on_delete=models.CASCADE)
+
+        class Meta:
+            abstract = True
+
+.. code-block:: python
+    :caption: ``production/models.py``
+
+    from django.db import models
+    from products.models import AbstractCar
+
+
+    class Manufacturer(models.Model):
+        pass
+
+
+    class Car(AbstractCar):
+        pass
+
+
+    # Car.manufacturer will point to `production.Manufacturer` here.
+
+.. admonition:: Blueprint models with relative references
+
+    Relative references allow you to define reusable blueprint models
+    with abstract classes, as relationships can resolve in concrete
+    subclasses across different applications.
```

Comment:
Thank you! May I suggest:
```suggestion
    Relative references enable the definition of reusable blueprint models with
    abstract classes, allowing relationships to resolve in concrete subclasses
    across different applications.
```

Reply:
I included these changes. Certainly an improvement on what was there before.

But I am not 100% satisfied that we're being as clear as we could be here. My understanding was that the suggestion meant for the docs to point out that because resolutions of relative references in different subclasses can point to different implementations of the pointed-to models, we can use them to create reusable "blueprint" models with relationships.

What do you think?